qiskit/_jobprocessor.py

```diff
@@ -0,0 +1,366 @@
+    for job in qobj['circuits']:
+        if job['compiled_circuit'] is None:
+            compiled_circuit = openquantumcompiler.compile(job['circuit'])
+            job['compiled_circuit'] = openquantumcompiler.dag2json(compiled_circuit)
+        sim_job = {'compiled_circuit': job['compiled_circuit'],
+                   'config': {**job['config'], **qobj['config']}}
+        local_simulator = simulators.LocalSimulator(
+            qobj['config']['backend'], sim_job)
+        local_simulator.run()
+        job_result['result'].append(local_simulator.result())
+        job_result['status'] = 'COMPLETED'
+    job_result['name'] = qobj['id']
+    job_result['backend'] = qobj['config']['backend']
+    return job_result
+
+def run_remote_backend(qobj, api, wait=5, timeout=60, silent=True):
+    """
+    Args:
+        qobj (dict): quantum object dictionary
+        api (IBMQuantumExperience): IBMQuantumExperience API connection
+
+    Raises:
+        QISKitError: if "ERROR" string in server response.
+    """
+    jobs = []
+    for job in qobj['circuits']:
+        if (('compiled_circuit_qasm' not in job) or
+            (job['compiled_circuit_qasm'] is None)):
+            compiled_circuit = openquantumcompiler.compile(job['circuit'].qasm())
+            job['compiled_circuit_qasm'] = compiled_circuit.qasm(qeflag=True)
+        if isinstance(job['compiled_circuit_qasm'], bytes):
+            jobs.append({'qasm': job['compiled_circuit_qasm'].decode()})
+        else:
+            jobs.append({'qasm': job['compiled_circuit_qasm']})
+
+    seed0 = qobj['circuits'][0]['config']['seed']
+    output = api.run_job(jobs, qobj['config']['backend'],
+                         shots=qobj['config']['shots'],
+                         max_credits=qobj['config']['max_credits'],
+                         seed=seed0)
+    if 'ERROR' in output:
+        raise QISKitError(output['ERROR'])
+    job_result = _wait_for_job(output['id'], api, wait=wait, timeout=timeout, silent=silent)
+    job_result['name'] = qobj['id']
+    job_result['backend'] = qobj['config']['backend']
+    return job_result
+
+def _wait_for_job(jobid, api, wait=5, timeout=60, silent=True):
+    """Wait until all online ran jobs are 'COMPLETED'.
+
+    Args:
+        jobid:  is a list of id strings.
+        api (IBMQuantumExperience): IBMQuantumExperience API connection
+        wait (int):  is the time to wait between requests, in seconds
+        timeout (int):  is how long we wait before failing, in seconds
+        silent (bool): is an option to print out the running information or
+            not
+
+    Returns:
+        A list of results that correspond to the jobids.
+
+    Raises:
+        QISKitError:
+    """
+    timer = 0
+    timeout_over = False
+    job_result = api.get_job(jobid)
+    if 'status' not in job_result:
+        from pprint import pformat
+        raise QISKitError("get_job didn't return status: %s" % (pformat(job)))
```

Comment:
In the same spirit as the comment in `_job_done_callback()`, is the `job` variable rechable?